pages/reset-password/index.tsx

```diff
@@ -0,0 +1,235 @@
+    clearErrors,
+  } = useForm<FormValues>({});
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
+    try {
+      // // Validate using Zod schema
+      await passwordSchema.parseAsync({
+        confirmPassword: data.confirmPassword,
+        password: data.password,
+      });
+
+      const response = await fetch(
+        "https://abeghelp-backend-staging.up.railway.app/api/v1/auth/password/reset",
+        {
+          method: "POST",
+          headers: {
+            "content-type": "application/json",
+          },
+          body: JSON.stringify({
+            token,
+            password: data.password,
+            confirmPassword: data.confirmPassword,
+          }),
+        },
+      );
+
+      if (response.ok) {
+        // Redirect or handle success
+        await new Promise<void>((resolve) => setTimeout(resolve, 2000));
+        await router.push("/reset-password/confirmation");
+      } else {
+        // Handle API errors
+        const responseData = (await response.json()) as ApiErrorResponse;
+        setServerError(responseData.message);
+        // Redirect to another page after 2 seconds
+        await new Promise<void>((resolve) => setTimeout(resolve, 2000));
+        await router.push("/signin");
+
+        // setTimeout(() => {
+        //   router.push("register");
+        // }, 2000);
+      }
+    } catch (error) {
+      if (error instanceof ZodError) {
+        // Extract error messages from ZodError
+        const validationErrors = Object.fromEntries(
+          error.errors.map((e) => [e.path.join("."), e.message]),
+        );
+
+        // Set errors for each field in the form
+        Object.keys(validationErrors).forEach((field) => {
+          // Ensure field is one of the allowed field names
+          if (field === "password" || field === "confirmPassword") {
+            setError(field as keyof FormValues, {
+              type: "manual",
+              message: validationErrors[field],
+            });
+          }
+        });
+      }
+    }
+  };
+
+  // Clear errors after a timeout
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+
+    if (formErrors.password || formErrors.confirmPassword || serverError) {
+      timeoutId = setTimeout(() => {
+        clearErrors();
+        setServerError("");
+      }, 3000);
+    }
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [
+    formErrors.password,
+    formErrors.confirmPassword,
+    serverError,
+    clearErrors,
+  ]);
+
+  return (
+    <main className="flex md:gap-10 lg:gap-10 xl:gap-0 px-5 md:px-20 h-[100dvh] w-full justify-center items-center bg-white">
+      <div className="hidden md:block md:w-3/5 lg:w-1/2">
+        <Image
+          src={"/abeg-plant.png"}
+          alt="abeg-plant"
+          width={500}
+          height={500}
+        />
+      </div>
+      <div className="md:w-1/2 lg:pr-10 xl:pr-20">
+        <form onSubmit={void handleSubmit(onSubmit)} className="flex flex-col">
```

Comment:
adding void here will render the function invalid and your form will still submit the default way. Do this instead

```
onSubmit={(event) => {
            event.preventDefault();
            void handleSubmit(onSubmit)(event);
          }}
```